packages/vscode-extension/src/project/metro.ts

```diff
@@ -292,6 +292,82 @@ export class Metro implements Disposable {
     ws.close();
   }
 
+  private async evaluateJsFromCdp(
+    webSocketDebuggerUrl: string,
+    source: string,
+    timeoutMs: number = 2000
+  ): Promise<string | undefined> {
+    const REQUEST_ID = 0;
+    let timeoutHandle: NodeJS.Timeout;
+
+    const wsUrl = new URL(webSocketDebuggerUrl);
+    wsUrl.hostname = "localhost";
+    wsUrl.port = String(this._port);
+
+    return new Promise((resolve, reject) => {
+      let settled = false;
+      const ws = new WebSocket(wsUrl);
+
+      timeoutHandle = setTimeout(() => {
+        Logger.debug(`[evaluateJsFromCdpAsync] Request timeout from ${wsUrl.toString()}`);
+        reject(new Error("Request timeout"));
+        settled = true;
+        ws.close();
+      }, timeoutMs);
+
+      ws.on("open", () => {
+        ws.send(
+          JSON.stringify({
+            id: REQUEST_ID,
+            method: "Runtime.evaluate",
+            params: { expression: source },
+          })
+        );
+      });
+
+      ws.on("error", (e) => {
+        Logger.debug(`[evaluateJsFromCdpAsync] Failed to connect ${wsUrl.toString()}`, e);
```

Comment:
Does this happen only when the connection fails? It's a bit misleading.

Reply:
we attach an error message with more details and it is true that we failed to connect